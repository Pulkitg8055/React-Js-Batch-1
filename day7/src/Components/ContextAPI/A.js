import React from "react";
import B from "./B";
export default function A({ handleClick }){
    return(
      <>
          <span> Hello from A </span>
          <br/>
          <B/>
          <button onClick={handleClick}> Click Me</button>

      </>
    );
}