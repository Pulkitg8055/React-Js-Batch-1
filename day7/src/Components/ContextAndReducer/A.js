import React, {useContext} from "react";
import {ApiContext} from "./ContextAndReducer";


export default function A(){
    const {state} = useContext(ApiContext);
    const DisplayUserNames = ({name}) => {
        return(
            <li>{name}</li>
        );
    };
    return(
        <ul>

            {state.usersData.map(user => <DisplayUserNames key={user.login} name={user.login}/>)}
        </ul>
    );
}