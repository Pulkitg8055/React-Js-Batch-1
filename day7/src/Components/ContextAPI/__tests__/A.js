import React from "react";
import {shallow} from "enzyme";
import A from "../A";
describe('Component A works correctly', () => {
    const handleClick = jest.fn();
    const componentA = shallow(<A handleClick={handleClick}/>);

    it('Component A renders correctly', () => {
        expect(componentA).toMatchSnapshot();
    });

    it('Button click works correctly', () =>{
        componentA.find('button').simulate('click');
        expect(handleClick).toHaveBeenCalled();
    });
});