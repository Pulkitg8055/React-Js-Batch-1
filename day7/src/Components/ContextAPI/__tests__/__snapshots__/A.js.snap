// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component A works correctly Component A renders correctly 1`] = `
<Fragment>
  <span>
     Hello from A 
  </span>
  <br />
  <B />
  <button
    onClick={[MockFunction]}
  >
     Click Me
  </button>
</Fragment>
`;
